src/windows/volume.cc

```diff
@@ -18,6 +18,32 @@
 
 static const size_t kVolumePathShortLength = 4;
 
+HRESULT drivelist::volume::GetSystemVolume(wchar_t *out) {
+  const char * const systemDrive = getenv("SystemDrive");
+  if (systemDrive == NULL)
+    return E_FAIL;
+  *out = systemDrive[0];
+  return S_OK;
+}
+
+HRESULT
+drivelist::volume::GetAvailableVolumes(std::vector<wchar_t> *const output) {
+  DWORD logicalDrivesMask = GetLogicalDrives();
+  if (logicalDrivesMask == 0)
+    return E_FAIL;
+
+  TCHAR currentDriveLetter = 'A';
+
+  while (logicalDrivesMask) {
+    if (logicalDrivesMask & 1)
+      output->push_back(currentDriveLetter);
+    currentDriveLetter++;
```

Comment:
We could, but I don't think its necessary. If we go past `Z`, `logicalDrivesMask` will be `false`, and thus the `while` loop will finish.